packages/eas-cli/src/build/android/build.ts

```diff
@@ -76,7 +76,7 @@ This means that it will most likely produce an AAB and you will not be able to i
     gradleContext
   );
   const versionCodeOverride =
-    ctx.easJsonCliConfig?.appVersionSource === AppVersionSource.REMOTE
+    ctx.easJsonCliConfig?.appVersionSource !== AppVersionSource.LOCAL
```

Comment:
I would stick to 
```suggestion
    ctx.easJsonCliConfig?.appVersionSource === AppVersionSource.REMOTE
```
because of the `--non-interactive` issues I highlighted in the previous comment